stan/math/prim/fun/to_complex.hpp

```diff
@@ -0,0 +1,44 @@
+#ifndef STAN_MATH_PRIM_FUN_TO_COMPLEX_HPP
+#define STAN_MATH_PRIM_FUN_TO_COMPLEX_HPP
+
+#include <stan/math/prim/meta.hpp>
+#include <type_traits>
+#include <complex>
+
+namespace stan {
+namespace math {
+
+/**
+ * Return a complex type from a real part and an imaginary part.
+ * Default values for both parts is 0.
+ *
+ * @tparam T type of real and
+ * @param re real element
+ * @param im imaginary element
+ * @return Complex type
+ */
+template <typename T = double, typename S = double>
```

Comment:
This templating is not right if `T` is an `int`.  The return type should always be floating point. 

The proper templating here is:

```c++
#include <stan/math/prim/meta/return_type.hpp>
...
template <typename T, typename S>
inline std::complex<stan::real_return_t<T, S>> to_complex(const T& re = 0, const S& im = 0) {
```

Sorry if the syntax isn't exactly right here.